packages/dds/merge-tree/src/sortedSegmentSet.ts

```diff
@@ -55,6 +59,28 @@ export class SortedSegmentSet<
     }
 
     private getOrdinal(item: T): string {
+        const maybeRef = item as Partial<LocalReferencePosition>;
+        if (maybeRef.getSegment !== undefined && maybeRef.isLeaf?.() === false) {
+            const lref = maybeRef as LocalReferencePosition;
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const segment = lref.getSegment()!;
+            const offset = lref.getOffset();
+            // Give the local reference a temporary yet valid ordinal.
+            // for this ordinal we treat each offset as a child position
+            // this is pretty cheap to compute, and i don't see any
+            // easy way to reuse, as the segment ordinal can change
+            // out of band
+            let previousOrdinal: string | undefined;
+            for (let i = 0; i <= offset; i++) {
+                previousOrdinal = computeOrdinal(
+                    segment.cachedLength,
```

Comment:
I think these values are insufficient with the strategy used for `computeOrdinal`: the input values to `String.fromCharCode` grow exponentially with `segment.cachedLength` and it seems like large enough inputs just return a null byte character (the MDN docs suggest [2^16 as the limit](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCharCode) which look right, and with some experimentation I'm guessing the implementation actually discards higher-order bits until numbers go out of int32 range).

Reply:
yup. the max char code is 0XFFFF . for numbers greater than that i need to simulate a tree to build up ordinal. i'm starting to think this might be more than it is worth, as i don't really care about the local refs being in ordinal order. i might just try to do something simpler, or maybe just go back to cramming them all under the segment ordinal